src/main/java/com/nylas/delta/DeltaQueryOptions.java

```diff
@@ -0,0 +1,90 @@
+package com.nylas.delta;
+
+import com.nylas.Maps;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Class representation of query parameters that can be passed in to Delta endpoints
+ * @see <a href="https://developer.nylas.com/docs/api/#tag--Deltas">Deltas</a>
+ */
+public class DeltaQueryOptions {
+
+	private String view;
+	private String include_types;
+	private String excluded_types;
+
+	public enum Type {
+		CONTACT("contact"),
+		FILE("file"),
+		MESSAGE("message"),
+		EVENT("event"),
+		DRAFT("draft"),
+		THREAD("thread"),
+		FOLDER("folder"),
+		LABEL("label"),
+
+		;
+
+		private final String name;
+
+		Type(String name) {
+			this.name = name;
+		}
+
+		public String getName() {
+			return name;
+		}
+	}
+
+	public DeltaQueryOptions setView(String view) {
+		this.view = view;
+		return this;
+	}
+
+	public DeltaQueryOptions includeTypes(Type... includeTypes) {
+		if(includeTypes.length > 0) {
+			this.include_types = Arrays.stream(includeTypes)
+					.map(Type::getName)
+					.collect(Collectors.joining(","));
+		}
+		return this;
+	}
+
+	public DeltaQueryOptions excludedTypes(Type... excludedTypes) {
+		if(excludedTypes.length > 0) {
+			this.excluded_types = Arrays.stream(excludedTypes)
+					.map(Type::getName)
+					.collect(Collectors.joining(","));
+		}
+		return this;
+	}
+
+	/**
+	 * Formats the Delta query options to a map
+	 * @return A mapping of Delta query options
+	 */
+	public Map<String, String> toMap() {
+		Map<String, String> map = new HashMap<>();
+		Maps.putIfNotNull(map, "view", view);
+		Maps.putIfNotNull(map, "include_types", include_types);
+		Maps.putIfNotNull(map, "excluded_types", excluded_types);
+		return map;
+	}
+
+	/**
+	 * Formats the Delta query options to a map if the query is valid
+	 * @return A mapping of Delta query options
+	 * @throws IllegalArgumentException if both {@code include_types} and {@code excluded_types} are set
+	 */
+	public Map<String, String> toValidMap() {
+		if(include_types != null && excluded_types != null) {
```

Comment:
any constraints for what a valid "view" could be?

Reply:
Good call out, view can only be expanded in the case of Deltas. Switching this to a boolean to avoid confusion, if the user wants an expanded view they can set it and we'll put it in the resulting map, else we don't include it.